todo_app/app.py

```diff
@@ -46,28 +49,39 @@ def load_user(name):
     return User(name)
 
 
-def check_writer_role(func):
-    @wraps(func)
-    def decorated_view(*args, **kwargs):
-        if login_manager._login_disabled:
-            return func(*args, **kwargs)
-        if current_user.id in WRITER_LIST:
-            return func(*args, **kwargs)
-        else:
-            return login_manager.unauthorized()
-    return decorated_view
-
-
 def create_app():
     app = Flask(__name__)
     app.config.from_object(Config())
 
+    app.logger.setLevel(app.config['LOG_LEVEL'])
+    if app.config['LOGGLY_TOKEN'] is not None:
+        handler = HTTPSHandler(f'https://logs-01.loggly.com/inputs/{app.config["LOGGLY_TOKEN"]}/tag/todo-app')
+        # Making sure werkzeug request logs are also sent to Loggly
+        getLogger('werkzeug').addHandler(HTTPSHandler(f'https://logs-01.loggly.com/inputs/{app.config["LOGGLY_TOKEN"]}/tag/todo-app-requests'))
+        formatter = jsonlogger.JsonFormatter()
+        handler.setFormatter(formatter)
+        app.logger.addHandler(handler)
+
     login_manager.init_app(app)
 
     mongo = MongoDB()
 
     client = WebApplicationClient(Config().GITHUB_CLIENT_ID)
 
+    def check_writer_role(func):
+        @wraps(func)
+        def decorated_view(*args, **kwargs):
+            if login_manager._login_disabled:
+                app.logger.warning("Application is running with login_manager._login_disabled = true")
+                return func(*args, **kwargs)
+            if current_user.id in WRITER_LIST:
+                app.logger.info("User %s is authorized" % current_user.id)
+                return func(*args, **kwargs)
+            else:
+                app.logger.error("User %s is unauthorized" % current_user.id)
```

Comment:
You probably want to reserve "error" logs for when there is a server-side error, not just a bad request - you can't prevent this happening and your application hasn't done anything wrong. So this could be a warning instead.

Then if you get lots of this warning, you can investigate because either someone is attacking the system, or you are probably providing a link/form to readers that you shouldn't.